plugins/tls.js

```diff
@@ -67,16 +68,50 @@ exports.tls_capabilities = function (next, connection) {
     if (connection.using_tls) { return next(); }
 
     var plugin = this;
-    
+
     if (plugin.cfg.no_tls_hosts[connection.remote_ip]) {
         return next();
     }
 
-    connection.capabilities.push('STARTTLS');
-    connection.notes.tls_enabled = 1;
+    var enable_tls = function () {
+        connection.capabilities.push('STARTTLS');
+        connection.notes.tls_enabled = 1;
+        next();
+    };
+
+    if (!plugin.cfg.redis || !server.notes.redis) {
+        return enable_tls();
+    }
+
+    var redis = server.notes.redis;
+    var dbkey = 'no_tls|' + connection.remote_ip;
 
-    /* Let the plugin chain continue. */
-    next();
+    redis.get(dbkey, function (err, dbr) {
+        if (err) {
+            connection.results.add(plugin, {err: err});
+            return enable_tls();
+        }
+
+        if (dbr === null) {
```

Comment:
Given that there's a PR to merge in ioredis instead of node-redis, you'd be better off doing this:  if (!dbr) otherwise this won't work if we change to ioredis later as it returns an empty string instead of null.